datafusion/physical-expr/src/utils/guarantee.rs

```diff
@@ -130,6 +124,39 @@ impl LiteralGuarantee {
             .fold(GuaranteeBuilder::new(), |builder, expr| {
                 if let Some(cel) = ColOpLit::try_new(expr) {
                     return builder.aggregate_conjunct(cel);
+                } else if let Some(inlist) = expr
+                    .as_any()
+                    .downcast_ref::<crate::expressions::InListExpr>()
+                {
+                    //Only support single-column inlist currently, multi-column inlist is not supported
+                    let col = inlist
+                        .expr()
+                        .as_any()
+                        .downcast_ref::<crate::expressions::Column>();
+                    if col.is_none() {
+                        return builder;
+                    }
```

Comment:
I think you could avoid the unwraps below by using something like this 

```suggestion
                    let Some(col) = col else {
                        return builder;
                    }
```

(and the same for `literals`)